aixplain/modules/pipeline/asset.py

```diff
@@ -144,11 +153,44 @@ def poll(self, poll_url: Text, name: Text = "pipeline_process") -> Dict:
                     resp["data"] = json.loads(resp["data"])["response"]
                 except Exception:
                     resp = r.json()
-            logging.info(f"Single Poll for Pipeline: Status of polling for {name} : {resp}")
+            logging.info(
+                f"Single Poll for Pipeline: Status of polling for {name} : {resp}"
+            )
         except Exception:
             resp = {"status": "FAILED"}
         return resp
 
+    def _should_fallback_to_v2(self, response: Dict, version: str) -> bool:
+        """Determine if the pipeline should fallback to version 2.0 based on the response.
+
+        Args:
+            response (Dict): The response from the pipeline call.
+            version (str): The version of the pipeline being used.
+
+        Returns:
+            bool: True if fallback is needed, False otherwise.
+        """
+        # If the version is not 3.0, no fallback is needed
+        if version != self.VERSION_3_0:
+            return False
+
+        should_fallback = False
+        if "status" not in response or response["status"] == "FAILED":
+            should_fallback = True
+        elif response["status"] == "SUCCESS" and (
+            "data" not in response or not response["data"]
```

Comment:
Should we also include when `len(response['data']) == 0`?

Reply:
`not response["data"]` already implies for empty lists also
```
if []:
    print("this wont happen")
```